tests/trx_generator/trx_provider.hpp

```diff
@@ -48,54 +67,38 @@ namespace eosio::testing {
 
       virtual ~provider_connection() = default;
 
-      void init_and_connect() {
-         _connection_thread_pool.start(
-             1, [](const fc::exception& e) { elog("provider_connection exception ${e}", ("e", e)); });
-         connect();
-      };
-
-      void cleanup_and_disconnect() {
-         disconnect();
-         _connection_thread_pool.stop();
-      };
-
-      fc::time_point get_trx_ack_time(const eosio::chain::transaction_id_type& _trx_id) {
-         fc::time_point time_acked;
-         std::lock_guard<std::mutex> lock(_trx_ack_map_lock);
-         auto search = _trxs_ack_time_map.find(_trx_id);
-         if (search != _trxs_ack_time_map.end()) {
-            time_acked = search->second;
-         } else {
-            elog("get_trx_ack_time - Transaction acknowledge time not found for transaction with id: ${id}", ("id", _trx_id));
-            time_acked = fc::time_point::min();
-         }
-         return time_acked;
-      }
+      void init_and_connect();
+      void cleanup_and_disconnect();
+      fc::time_point get_trx_ack_time(const eosio::chain::transaction_id_type& trx_id);
+      void trx_acknowledged(const eosio::chain::transaction_id_type trx_id, const fc::time_point ack_time);
```

Comment:
Is there a reason you are using `const` instead of `const&`? Certainly trx_id should be `const&`. `time_point` could be either `const&` or by value.